docs/en/testing/supplemental.md

```diff
@@ -158,6 +123,187 @@ describe('MyWidget', () => {
+		h.expect(() => <div key="root">{`{"width":100}`}</div>);
+	});
+});
+```
+
+##### Mock `Store` Middleware
+
+Using `createMockStoreMiddleware` from `@dojo/framework/testing/mocks/middleware/store` creates a typed mock store middleware, which optionally supports mocking processes. To mock a store process pass a tuple of the original store process and the stub process. The middleware will swap out the call to the original process for the passed stub. If no stubs are passed, the middleware will simply no-op all process calls.
+
+To make changes to the mock store, call the `mockStore` with a function that returns an array of store operations. This is injected with the stores `path` function, that should be used to create the pointer to the state that needs changing.
+
+```tsx
+mockStore((path) => [replace(path('details', { id: 'id' })]);
+```
+
+Consider the following widget:
+
+> src/MyWidget.tsx
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom'
+import { myProcess } from './processes';
+import MyState from './interfaces';
+// application store middleware typed with the state interface
+// Example: `const store = createStoreMiddleware<MyState>();`
+import store from './store';
+
+const factory = create({ store }).properties<{ id: string }>();
+
+export default factory(function MyWidget({ properties: { id }, middleware: store }) {
+    const { path, get, executor } = store;
+    const details = get(path('details');
+    let isLoading = get(path('isLoading'));
+
+    if ((!details || details.id !== id) && !isLoading) {
+        executor(myProcess)({ id });
+        isLoading = true;
+    }
+
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    return <ShowDetails {...details} />;
+});
+```
+
+Using the mock store middleware would look like:
+
+> tests/unit/MyWidget.tsx
+
+```tsx
+import { tsx } from '@dojo/framework/core/vdom'
+import createMockStoreMiddleware from '@dojo/framework/testing/mocks/middleware/store';
+import harness from '@dojo/framework/testing/harness';
+
+import { myProcess } from './processes';
+import MyWidget from './MyWidget';
+import MyState from './interfaces';
+import store from './store';
+
+// import a stub/mock lib, doesn't have to sinon
+import { stub } from 'sinon';
+
+describe('MyWidget', () => {
+     it('test', () => {
+          const properties = {
+               id: 'id'
+          };
+         const myProcessStub = stub();
+         // type safe mock store middleware
+         // pass through an array of tuples `[originalProcess, stub]` for mocked processes
+         // calls to processes not stubbed/mocked will just be ignored
+         const mockStore = createMockStoreMiddleware<MyState>([[myProcess, myProcessStub]]);
+         const h = harness(() => <MyWidget {...properties} />, {
+             middleware: [store, mockStore]
+         });
+         h.expect(/* assertion template for `Loading`*/);
+
+         // would be useful to assert the calls to processes effectively stubbed
```

Comment:
This comment seems a bit weird to be in the docs as it feels more like an internal todo

Reply:
old comment, good catch.